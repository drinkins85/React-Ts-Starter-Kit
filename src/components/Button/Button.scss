.Button {
    background-color: darkorange;
    border-radius: 10px;
    background-image: url("./assets/time.svg");
    background-repeat: no-repeat;
    user-select: none;
}
