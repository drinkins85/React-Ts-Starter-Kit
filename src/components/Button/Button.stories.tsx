import * as React from 'react';
import Button from './Button';

export default {
    title: 'Button',
    component: Button,
    argTypes: {
        color: { control: 'color' },
        size: {
            control: {
                type: 'range',
                min: 6,
                max: 50,
            },
        },
    },
};

export const Base = (args) => <Button {...args} />;
Base.args = {
    title: 'test',
};
