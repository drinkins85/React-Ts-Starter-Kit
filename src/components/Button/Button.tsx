import * as React from 'react';
import './Button.scss'
import TimePic from './assets/time.png';

type TitleProps = {
    title: string;
    color?: string;
    size?: number;
};

function Button({ title, color, size }: TitleProps): JSX.Element {
    const onTitleClick = () => {
        console.log('test');
    };

    const styles:React.CSSProperties = {
        color,
        fontSize: size,
    };

    return (
        <button
            type="button"
            className="Button"
            onClick={onTitleClick}
            style={styles}
        >
            {title}
            <img src={TimePic} className="Title-Image" alt="" />
        </button>
    );
}

export default Button;
