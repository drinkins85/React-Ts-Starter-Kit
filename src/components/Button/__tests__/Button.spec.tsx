/// <reference types="jest" />
import * as React from 'react';
import { render } from 'enzyme';
import Button from '../Button';

it('show Button', () => {
    const title = 'test';
    const result = render(<Button title={title} />);

    expect(result).toMatchSnapshot();
});
