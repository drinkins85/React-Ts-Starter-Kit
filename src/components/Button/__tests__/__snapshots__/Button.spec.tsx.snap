// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show Button 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "Button",
      "type": "button",
    },
    "children": Array [
      Object {
        "data": "test",
        "next": Object {
          "attribs": Object {
            "alt": "",
            "class": "Title-Image",
          },
          "children": Array [],
          "name": "img",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "alt": undefined,
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "alt": undefined,
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
      Object {
        "attribs": Object {
          "alt": "",
          "class": "Title-Image",
        },
        "children": Array [],
        "name": "img",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "data": "test",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "alt": undefined,
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "alt": undefined,
          "class": undefined,
        },
      },
    ],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "type": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "type": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
